core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -98,6 +98,8 @@ private[spark] object Utils extends Logging {
   /** Scheme used for files that are locally available on worker nodes in the cluster. */
   val LOCAL_SCHEME = "local"
 
+  private val PATTERN_FOR_COMMAND_LINE_ARG = "-D(.+)=(.+)".r
```

Comment:
First group should be non-greedy, otherwise I'm almost sure that `-Dfoo=bar=baz` will do the wrong matching.

Reply:
Nice finding! Will address.